### PR TITLE
Clean up `SiteConfig`

### PR DESCRIPTION
**Pull request by citizen428**
## What type of PR is this? (check all applicable)

- [X] Cleanup

## Description

Another exciting PR from the settings refactoring department, aka me. This PR removes now obsolete settings from `SiteConfig`. It also cleans up all previously forgotten mentions of `SiteConfig` or those that were reverted by merges. Now that the fields are gone from `SiteConfig` these became a lot easier to spot since they now raise `NoMethodError`s. Additionally some obsolete data update scripts also got removed. This should only be merged once all the DUS ran across the fleet, so I'll assign it to you, @Zhao-Andy.

## Related Tickets & Documents

https://github.com/forem/rfcs/blob/0821fad5a640427c54dbbfe36b3389ba0d7dbe4c/text/0126-split-site-config.md

## QA Instructions, Screenshots, Recordings

n/a

### UI accessibility concerns?

n/a

## Added tests?

- [X] No, and this is why: they were added during the original refactoring, this only removes the obsolete config settings

## [Forem core team only] How will this change be communicated?

- [X] This change does not need to be communicated, and this is why not: nothing worth communicating here.

### CODE REVIEW

**Comment by citizen428** [comment]
Note: the second commit is only here temporarily until #13739 gets merged into `main`.

**Review by rhymes — DISMISSED** [review]

**Comment by rhymes** [comment]
@citizen428 there are a bunch of specs that have failed that need to be migrated. I'll wait to approve right after

**Comment by citizen428** [comment]
@rhymes Yes, what a mess 😞  Some were clearly forgotten when I did the original work, some changes seem to have been undone by merges 😭

**Comment by citizen428** [comment]
Note: this PR now temporarily disables all specs for `spec/requests/api/v0/admin/configs_spec.rb`. Since the API controller itself isn't properly working this should be ok and the specs will come back once the controller has been refactored.

**Review by rhymes — APPROVED** [review]

File: lib/data_update_scripts/20210512033821_clean_up_site_config.rb
Comment (djuber) [inline lib/data_update_scripts/20210512033821_clean_up_site_config.rb:52]: This is okay now because the data updates moving these values _already_ happened *AND* nothing was writing to these fields since we moved/copied the values?

**Review by djuber — APPROVED** [review]
I was a little cautious about reviewing a PR with 58 changed files, but apart from the data update scripts this is almost entirely removal replacement of constant names and removal of now dead code. My question about sequencing is almost entirely due to not reading more carefully the rest of the commits, basically if we're deleting SiteConfig values the time from copying them into the Settings and only reading/writing to the Settings module should be as short as possible (to prevent a fresh write after copy leaving stale choices, but I expect those will be minimal).

This was a pleasure to review because it's almost all 1-line per file substitutions!

File: lib/data_update_scripts/20210512033821_clean_up_site_config.rb
Comment (djuber) [reply lib/data_update_scripts/20210512033821_clean_up_site_config.rb:52]: Totally *non-blocking* and I'm likely creating complications out of thin air, but the idea that we know whether and when people have upgraded code has a shelf life.

While we probably still control _most_ of the supported forem instances, is there any real risk to a self-hosting pioneer trying to update in the future (someone who had started a Forem last week while SiteConfig was still a location for much of the configuration, and attempts to update to the next release after this is merged and released)? 

Is the worst case scenario that an update removes configuration that _didn't_ also already get copied elsewhere? Or is this (because of in-order execution) guaranteed to run _after_ the other update scripts. Is this some chasm where standing on one side you have a working instance, and updating to the other side where scripts have been deleted from the repo/filesystem and thus will _never_ run you have a non-functioning system, or is it only a crack where the outcome is a poorly configured system (you can still login to admin and set things up, but you might have lost some of your config choices)?

Comment (citizen428) [reply lib/data_update_scripts/20210512033821_clean_up_site_config.rb:52]: > This is okay now because the data updates moving these values already happened AND nothing was writing to these fields since we moved/copied the values?

@djuber I may be missing a case or two, but generally, updates of values always went to the right tables (the `settings_*` ones) though we still showed information from `SiteConfig` in several places which means updates wouldn't have been properly reflected. 

> is there any real risk to a self-hosting pioneer trying to update in the future (someone who had started a Forem last week while SiteConfig was still a location for much of the configuration, and attempts to update to the next release after this is merged and released)?

Not if they ran all data update scripts.

> Or is this (because of in-order execution) guaranteed to run after the other update scripts. 

I wanna say yes, but we are talking about software so I'll say "to a reasonable degree of certainty" instead.

>  Is this some chasm where standing on one side you have a working instance, and updating to the other side where scripts have been deleted from the repo/filesystem and thus will never run you have a non-functioning system

See above. Lately, when we deleted scripts we tried to also always remove the script itself (`DataUpdateScript.delete_bu(name: "...")`) so hopefully we are good on that front.

Thanks for bringing up these points, they are are very valuable and we should talk about these things more. In this specific instance I will take a leap of faith and hope for the best because I really need to unblock myself to be able to finish the work on this RFC. 🤞

**Pull request merged**
